opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -990,15 +990,17 @@ static void mca_btl_base_am_process_atomic(mca_btl_base_module_t *btl,
     switch (hdr->type) {
     case MCA_BTL_BASE_AM_ATOMIC:
         if (4 == hdr->data.atomic.size) {
-            uint32_t tmp = (uint32_t) atomic_response;
```

Comment:
There are many inconsistencies in this function between the use of the target_address pointer in the MCA_BTL_BASE_AM_ATOMIC and MCA_BTL_BASE_AM_CAS branches. The `if` branch with with size 8 should be an 'else if'

Reply:
FWIW, this PR was only intended to be about fixing compiler warnings.  Do you want code refactorings here on this PR, too?

Reply:
I would prefer to keep things separate, but as a result of this PR all `if` branches in this function have different casting to fix to the same issue. Can we at least have the same fix as in the other branches ?

Reply:
@bosilca and I chatted about this on the phone.  I pushed up a revision that I think addresses everything.